resources/lib/upgrade_controller.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""Check if the addon has been updated and make necessary changes"""
+from __future__ import unicode_literals
+
+import resources.lib.common as common
+
+from resources.lib.globals import g
+from resources.lib.database.db_update import run_local_db_updates, run_shared_db_updates
+
+
+def check_addon_upgrade():
+    """Check addon upgrade and perform necessary update operations"""
+    addon_previous_ver = g.LOCAL_DB.get_value('addon_previous_version', g.VERSION)
```

Comment:
If there's no 'addon_previous_version' value in the db, then previous version = g.VERSION (so the current version)